src/query/src/part_sort.rs

```diff
@@ -418,15 +481,64 @@ impl PartSortStream {
             )
         })?;
 
-        // only clear after sorted for better debugging
-        self.buffer.clear();
         self.produced += sorted.num_rows();
```

Comment:
Looks like removing the `buffer.clear()`, any considerations about it?

Reply:
`clear` is moved to the front (it's `std::mem::replace` now)